src/k8s-extension/azext_k8s_extension/partner_extensions/ContainerInsights.py

```diff
@@ -473,6 +475,26 @@ def _get_container_insights_settings(cmd, cluster_resource_group_name, cluster_r
             logger.info("provided useAADAuth flag is : %s", useAADAuthSetting)
             if (isinstance(useAADAuthSetting, str) and str(useAADAuthSetting).lower() == "true") or (isinstance(useAADAuthSetting, bool) and useAADAuthSetting):
                 useAADAuth = True
+        if useAADAuth and ('dataCollectionSettings' in configuration_settings):

```

Comment:
>'dataCollectionSettings' [](http://example.com/codeflow?start=27&length=24)

The Python in operator is case sensitive.  Are you sure the value will always be in camel case?  If not, you'd want to handle different cases

Reply:
> > 'dataCollectionSettings'
> 
> The Python in operator is case sensitive. Are you sure the value will always be in camel case? If not, you'd want to handle different cases

> yes, this will be always camel case. This is contract across the client and Monitoring RP.